File: src/main/java/com/braydenoneal/blang/testing/test/WhileLoops.java
package com.braydenoneal.blang.testing.test;

import com.braydenoneal.blang.parser.expression.value.IntegerValue;

import java.util.List;

public class WhileLoops extends Test {
    @Override
    public String body() {
        return """
                i = 0;
                
                while i < 10 {
                    i += 1;
                }
                """;
    }

    @Override
    public List<Expect> expects() {
        return List.of(
                new Expect("i", new IntegerValue(10))
        );
    }
}

File: src/main/kotlin/com/braydenoneal/blang/testing/test/WhileLoops.kt
package com.braydenoneal.blang.testing.test

import com.braydenoneal.blang.parser.expression.value.IntegerValue

class WhileLoops : Test() {
    override fun body(): String {
        return """
                i = 0;
                
                while i < 10 {
                    i += 1;
                }
                """.trimIndent()
    }

    override fun expects(): List<Expect> {
        return listOf(
            Expect("i", IntegerValue(10))
        )
    }
}
